processing/data_collection/gazette/spiders/ma_sao_luis.py

```diff
@@ -0,0 +1,106 @@
+  -- Wait until a selector matches an element
+  -- and it contains a specific value
+  -- in the page. Return an error if waited more
+  -- than maxwait seconds.
+  if maxwait == nil then
+      maxwait = 10
+  end
+  return splash:wait_for_resume(string.format([[
+    function main(splash) {
+      var selector = '%s';
+      var value = %s;
+      var maxwait = %s;
+      var end = Date.now() + maxwait*1000;
+      function check() {
+        if(document.querySelector(selector) && document.querySelector(selector).value === value) {
+          splash.resume('Element found');
+        } else if(Date.now() >= end) {
+          var err = 'Timeout waiting for element';
+          splash.error(err + " " + selector);
+        } else {
+          setTimeout(check, 200);
+        }
+      }
+      check();
+    }
+  ]], css, value, maxwait))
+end
+
+function main(splash, args)
+    getElementByXPath = splash:jsfunc([[
+    function (path) {
+        return document.evaluate(path, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue;
+    }
+    ]])
+    splash:go(args.url)
+    wait_for_element(splash, 'a.campoResultadoDownload', '1', 40)
+    page_input_box = getElementByXPath("//div[@class='GPFMNGWNP']/input[@class='gwt-TextBox GPFMNGWOJ GPFMNGWIK']")
+    page_input_box:send_keys("<Delete>")
+    page_input_box:send_text(args.page_number)
+    page_input_box:send_keys("<Return>")
+    wait_for_element(splash, 'a.campoResultadoDownload', args.page_number, 40)
+    return {html = splash.html(), cookies = splash:get_cookies(), png = splash.png()}
+end
+"""
+
+    def parse(self, response):
+        """
+        @url
+        @returns items 1
+        @scrapes date file_urls is_extra_edition
+                 municipality_id power scraped_at
+        """
+        self.logger.info("sent page %d to splash", 1)
+        yield SplashRequest(
+            url=response.url, callback=self.parse_response,
+            endpoint='execute', args={'lua_source': self.lua_script,
+                                      'page_number': '1', 'timeout': 90})
+
+    def parse_response(self, response):
+        for gazette_table in response.xpath(
+                "//table[contains(@class, 'tabelaResultado')]"):
+            date = gazette_table.xpath(
+                    ".//tbody/tr[1]/td[2]/div/text()").extract_first()
+            self.logger.info("ma_sao_luis_spider: got gazette for day %s",
+                             date)
+            date = dateparser.parse(date, settings={'DATE_ORDER': 'DMY'})
+
+            if date.year < 2015:
+                print('ano de 2015, parando. data: {}'.format(date))
+                raise StopIteration
+            url = gazette_table.xpath((".//tbody/tr/td/a"
+                                       "[contains(text(), 'Download')]"
+                                       "/@href")).extract_first()
+            url = response.url[:39] + url
+            extra_edition = (gazette_table.xpath(
```

Comment:
I don't like putting things inside parenthesis. If for some mistake we include a comma, it will be a bug very difficult to find:
```
In [1]: bool((False))
Out[1]: False

In [2]: bool((False, ))
Out[2]: True
```